app/models/story_parser.rb

```diff
@@ -246,6 +225,19 @@ def parse_author(location,external_author_name,external_author_email)
 
   protected
 
+  # tries to create an external author for a given url
+  def parse_author(location, external_author_name, external_author_email)
+    if location.present? && external_author_name.blank? && external_author_email.blank?
+      source = get_source_if_known(KNOWN_AUTHOR_PARSERS, location)
+      if source.nil?
+        raise Error, "No external author name or email specified, and unable to generate email based on source location"
+      else
+        eval("parse_author_from_#{source.downcase}(location)")
```

Comment:
The evals will need to hang around a little longer until I extract and refactor the code for specific external sites, at which point I might be able to remove them.